tests/system_tests/experiment_plans/test_plan_system.py

```diff
@@ -20,18 +20,17 @@ def test_getting_data_for_ispyb():
     attenuator = i03.attenuator(fake_with_ophyd_sim=True)
     flux = i03.flux(fake_with_ophyd_sim=True)
     dcm = i03.dcm(fake_with_ophyd_sim=True)
-
-    undulator.wait_for_connection()
-    synchrotron.wait_for_connection()
-    slit_gaps.wait_for_connection()
-    attenuator.wait_for_connection()
-    flux.wait_for_connection()
+    robot = i03.robot(fake_with_ophyd_sim=True)
 
     RE = RunEngine()
 
     @bpp.run_decorator()
-    def standalone_read_hardware(und, syn, slits, att, flux):
-        yield from read_hardware_for_ispyb_pre_collection(und, syn, slits)
+    def standalone_read_hardware(und, syn, slits, robot, att, flux):
+        yield from read_hardware_for_ispyb_pre_collection(und, syn, slits, robot)
         yield from read_hardware_for_ispyb_during_collection(att, flux, dcm)
 
-    RE(standalone_read_hardware(undulator, synchrotron, slit_gaps, attenuator, flux))
+    RE(
+        standalone_read_hardware(
+            undulator, synchrotron, slit_gaps, robot, attenuator, flux
+        )
+    )
```

Comment:
We don't seem to assert anything here

Reply:
Agreed it's crap - spun into https://github.com/DiamondLightSource/hyperion/issues/1129